### PR TITLE
[#15046] Contacts in contact list are sorted by 3-random name

### LINKED ISSUE #15046: Contacts in contact list are sorted by 3-random name
# Bug Report

## Problem 
image

#### Expected behavior
Contacts are sorted bu name if it is set 

### Reproduction

1) Open Status
2) Add several mutual contacts from desktop client

### Additional Information

- Status version: rc-1 
- Operating System: Android, iOS

[comment]: # (Please, add logs/notes if necessary)

### PR DESCRIPTION
**Pull request by flexsurfer**
fixes #15046

### CODE REVIEW

File: src/status_im/multiaccounts/core.cljs [hunk 1]
```diff
@@ -56,17 +53,17 @@
...
+      (take 2 non-empty-names)
+      [(first non-empty-names)
+       (when public-key? (utils/get-shortened-address (or compressed-key public-key)))])))
 
 (defn contact-with-names
   "Returns contact with :names map "
   [contact]
-  (assoc contact :names (contact-names contact)))
+  (let [contact' (assoc contact :names (contact-names contact))]
+    (assoc contact' :two-names (contact-two-names contact' true))))
```
Comment (ibrkhalil) [inline src/status_im/multiaccounts/core.cljs:66]: Can this be one `assoc` ?

**Review by ibrkhalil — APPROVED** [review]
Non blocking comment

File: src/status_im/multiaccounts/core.cljs (hunk 1 above)
Comment (flexsurfer) [reply src/status_im/multiaccounts/core.cljs:66]: no, yeah i know its ugly

Comment (flexsurfer) [reply src/status_im/multiaccounts/core.cljs:66]: but it needs a general refactoring, this is a hotfix

**Review by OmarBasem — APPROVED** [review]

**Comment by churik** [comment]
Please, cherry-pick it to release/1.22.x branch.

Thank you for fix!

**Pull request merged**